lib/logreader.c

```diff
@@ -215,8 +214,10 @@ log_reader_io_process_input(gpointer s)
        */
       if (!main_loop_worker_job_quit())
         {
+          log_pipe_ref(&self->super.super);
           log_reader_work_perform(s);
           log_reader_work_finished(s);
+          log_pipe_unref(s);
         }
```

Comment:
I don't think it is necessary to take + drop a reference here, unless I am missing something. 

We have a borrowed reference due to our argument. Also, should it still be needed there's an asymmetry 
between the ref/unref call, one taking self->super.super, the other taking an untyped "s".

Reply:
Good point, I do not see any reason either. I just blindly copied the ref/unref to maintain original behavior, but I agree these are not necessary in the nonthreaded variant. I will remove these.